edx_exams/apps/access/views.py

```diff
@@ -0,0 +1,63 @@
+"""
+Exam Access Ticket Views
+"""
+from datetime import datetime, timedelta
+
+from django.http import HttpResponse
+from rest_framework import status
+from rest_framework.views import APIView
+from token_utils.sign import create_jwt
+
+from edx_exams.apps.core.models import ExamAttempt
+
+
+class ExamAccessTicketsView(APIView):
+    """
+    todo: add docs after impl is complete
+    """
+
+    # todo: should this be a get request?
+    def get(self, request):
+        user = request.user
+        exam = request.exam
+        claims = {}
+        claims.update(exam.course_id)
+        claims.update(exam.content_id)
+        exp_secs = 60
+
+        exam_attempt = ExamAttempt.get_latest_user_exam_attempt(user, exam)
+        # todo: should this be called ticket or token? For consistency?
+        cookie_name = "exam_access_ticket"
+
+        # If exam attempt exists for user, then grant exam access
+        if exam_attempt is not None:
+            # Grant access based on a short check-in window or the time to the end of the exam attempt,
+            # whichever is shorter.
+            if exam_attempt.start_time is not None and exam_attempt.allowed_time_limit_mins is not None:
```

Comment:
Question: does this logic make sense? I need to fix the datetime.now() on line 38 with some debugging (because of data type inconsistencies), but I just want to check that the logic overall is sound. Also curious if anyone has feedback on better ways to approach this.

Reply:
this whole block is logic we'll definitely need to reuse in other parts of this app. Maybe create a utility function for `get_attempt_time_remaining`? This should probably just return 0 if we're past the end time.

It looks like right now exp_secs is becoming negative if we're beyond the end time? How do you think we should handle that?